### DISCUSSION TITLE
How to install package when pyproject.toml is inside the package

### DISCUSSION THREAD

**Discussion by m-o-leary** [python-poetry/poetry#7862] [Q&A]
I am trying to migrate a legacy project to use `poetry`. The structure of the project is causing me problems though.

The project is currently setup like this (after adding in `poetry` on my feature branch):
```
package
├── Makefile
├── README.md
├── module_1
├── module_2
├── module_3
├── poetry.lock
├── pyproject.toml
├── module_4
├── module_5
└── module_6
```
The project is legacy as I mentioned and while I understand the structure should be like:
```
package
├── Makefile
├── README.md
├── poetry.lock
├── pyproject.toml
└──  package
    ├── module_1
    ├── module_2
    ├── module_3
    ├── module_4
    └── module_5
```
If I do make that change on my dev branch, I can't merge in changes from `main` so that is not an option while this structure remains on `main` (~ few weeks at least)

 I've looked at the docs [here](https://python-poetry.org/docs/pyproject/#packages) but can't seem to figure out how to install all the modules and have them under the namespace `package`.
 
 If I run `poetry install` I get the message:
 ```
Installing the current project: package (0.1.0)
```

However when I try to import any of my modules it fails:
```
Python 3.7.16 (default, Mar 23 2023, 17:48:55) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from package import module_1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'package'
```

Does poetry have the machinery to handle my case?

**Comment by ralbertazzi** [comment]
I'm not sure there's a way to achieve what you want with that structure. You mentioned the right docs, which should be the way to go to install that kind of project structure. However, you will end up with module1 / module2 / .. as root packages
